test/core/iomgr/resolve_address_test.cc

```diff
@@ -428,6 +428,70 @@ TEST_F(ResolveAddressTest, CancelWithNonResponsiveDNSServer) {
   PollPollsetUntilRequestDone();
 }
 
+// RAII class for pollset and pollset_set creation
+class PSS {
+ public:
+  static std::unique_ptr<PSS> Create() {
+    return absl::WrapUnique<PSS>(new PSS());
+  }
+
+  ~PSS() {
+    grpc_pollset_set_del_pollset(pss_, ps_);
+    grpc_pollset_destroy(ps_);
+    gpr_free(ps_);
+    grpc_pollset_set_destroy(pss_);
+    gpr_log(GPR_DEBUG, "PSS:%p deleted", this);
+  }
+
+  grpc_pollset_set* pollset_set() { return pss_; }
+
+ private:
+  PSS() {
+    ps_ = static_cast<grpc_pollset*>(gpr_zalloc(grpc_pollset_size()));
+    grpc_pollset_init(ps_, &mu_);
+    pss_ = grpc_pollset_set_create();
+    grpc_pollset_set_add_pollset(pss_, ps_);
+    gpr_log(GPR_DEBUG, "PSS:%p created", this);
+  }
+
+  gpr_mu* mu_;
+  grpc_pollset* ps_;
+  grpc_pollset_set* pss_;
+};
+
+TEST_F(ResolveAddressTest, DeleteInterestedPartiesAfterCancellation) {
+  // Regression test for race around interested_party deletion after
+  // cancellation.
+  if (std::string(g_resolver_type) != "ares") {
```

Comment:
Nit: Could use `absl::string_view` instead of `std::string` here to avoid an unnecessary allocation.

Reply:
Done.